connection.go

```diff
@@ -127,33 +166,35 @@ func (c *conn) prepareStmt(cmd string) (*stmt, error) {
 	return &stmt{c: c, stmt: &s}, nil
 }
 
-func (c *conn) execUnprepared(cmd string) (driver.Result, error) {
-	cmdstr := C.CString(cmd)
-	defer C.free(unsafe.Pointer(cmdstr))
+func (c *conn) extractStmts(query string) (C.duckdb_extracted_statements, C.idx_t, error) {
+	cquery := C.CString(query)
+	defer C.free(unsafe.Pointer(cquery))
 
-	var res C.duckdb_result
-	err := C.duckdb_query(*c.con, cmdstr, &res)
-	defer C.duckdb_destroy_result(&res)
+	var stmts C.duckdb_extracted_statements
+	var stmtsCount C.idx_t
+	stmtsCount = C.duckdb_extract_statements(*c.con, cquery, &stmts)
+	err := C.GoString(C.duckdb_extract_statements_error(stmts))
 
-	if err == C.DuckDBError {
-		dbErr := C.GoString(C.duckdb_result_error(&res))
-		return nil, errors.New(dbErr)
+	if err != "" {
+		C.duckdb_destroy_extracted(&stmts)
+		return nil, 0, errors.New(err)
+	}
+	if stmtsCount == 0 {
+		C.duckdb_destroy_extracted(&stmts)
+		return nil, 0, errors.New("no statements found")
 	}
```

Comment:
The docs say that `stmtsCount == 0` indicates failure. They don't say that `duckdb_extract_statements_error` is safe to call if the query didn't fail. So probably restructure this to ensure safe handling in accordance with the docs.

Reply:
> The docs say that stmtsCount == 0 indicates failure.

I looked at the documentation, I am not sure where it explicitly says that. Even in one of the test cases [here](https://github.com/duckdb/duckdb/blob/master/test/api/capi/test_capi_extract.cpp#L48) it calls `duckdb_extract_statements_error` even if query didn't fail, so thought its ok to call in every case. Anyways I think it would be safe to do this ?
```
	if stmtsCount == 0 {
		err := C.GoString(C.duckdb_extract_statements_error(stmts))
		C.duckdb_destroy_extracted(&stmts)

		if err != "" {
			return nil, 0, errors.New(err)
		}
		return nil, 0, errors.New("no statements found")
	}
```
Let me know what you think ?